### ISSUE TITLE
CTP6 /  CoreClr beta3 Support

### ISSUE THREAD

**Issue by sandorfr** [Azure/azure-storage-net#110]
Everything was working fine with beta2 but it looks like visualstudio CTP6 / CoreClr beta3 isn't supported at the moment. 

An ExecuteQuerySegmentedAsync fails with message 
```
Attempt by security transparent method 'Microsoft.Data.OData.ODataMessageReaderSettings.get_BaseUri()' to access security critical type 'System.Uri' failed.
```

Currently the only way to use this package with is to run with the full .net framework (which is not the point when building something on top of vnext :))

**Comment by sandorfr** [comment]
It throws in TableOperationHttpRequestMessageFactory.cs at line 84 when it calls ODataMessageWriter constructor.

I tried to find source code for Microsoft.Data.OData 5.6.3 but It looks like it is not available.

**Comment by pemari-msft** [comment]
Thank you for reporting this issue. Like you identified, this error is being thrown by our OData dependency. However, we’ve done some investigation and have determined that this is actually a bug in CoreCLR Beta3 (CTP 6). We’ve reported this to the CoreCLR team and they’ve assured us they will fix it in an upcoming build. We apologize for the inconvenience.

**Comment by sandorfr** [comment]
Thanks for the update.

**Comment by toothbrush7777777** [comment]
@pemari-msft  I've got the exact same problem.
Do you have any idea of when a fix might be available?

**Comment by pemari-msft** [comment]
Unfortunately, no. The CoreCLR team is working on a fix but we don't have an ETA at this time. Using CTP4 (beta1 or beta2) should be working correctly if you would like to use that as a temporary workaound.

**Comment by pemari-msft** [comment]
This is now fixed -- try using version 4.4.1-preview of our library on nuget (https://www.nuget.org/packages/WindowsAzure.Storage/) and let us know if you have any further issues. Thanks!

**Comment by sandorfr** [comment]
Thanks for the update I'll try this new version this week.
